pkg/models/load_balancer.go

```diff
@@ -209,27 +284,133 @@ type CreateLBProfile struct {
 
 type CreateLBProfileReq struct {
 	ID            int       `json:"-" tf:"id,computed"`
-	Name          string    `json:"name"`
-	Description   string    `json:"description"`
+	LbID          int       `json:"-" tf:"lb_id"`
+	Name          string    `json:"name" tf:"name"`
+	Description   string    `json:"description" tf:"description"`
 	ServiceType   string    `json:"serviceType" tf:"service_type"`
-	ProfileConfig LBProfile `json:"config"`
+	ProfileConfig LBProfile `json:"config"  tf:"config"`
 }
 
 type LBProfile struct {
-	ProfileType            string `json:"profileType" tf:"profile_type"`
-	RequestHeaderSize      int    `json:"requestHeaderSize" tf:"request_header_size"`
-	ResponseHeaderSize     int    `json:"responseHeaderSize" tf:"response_header_size"`
-	ResponseTimeout        int    `json:"responseTimeout" tf:"response_timeout"`
-	HTTPIdleTimeoutName    int    `json:"httpIdleTimeout" tf:"http_idle_timeout"`
-	FastTCPIdleTimeout     int    `json:"fastTcpIdleTimeout" tf:"fast_tcp_idle_timeout"`
-	SSLSuite               string `json:"sslSuite" tf:"ssl_suite"`
+	TfHttpConfig    *CreateHttpProfileConfig    `json:"-" tf:"http_monitor,sub"`
+	TfTcpConfig     *CreateTcpProfileConfig     `json:"-" tf:"tcp_monitor,sub"`
+	TfUdpConfig     *CreateUdpProfileConfig     `json:"-" tf:"udp_profile,sub"`
+	TfCookieConfig  *CreateCookieProfileConfig  `json:"-" tf:"cookie_profile,sub"`
+	TfGenericConfig *CreateGenericProfileConfig `json:"-" tf:"generic_profile,sub"`
+	TfSourceConfig  *CreateSourceProfileConfig  `json:"-" tf:"sourceip_profile,sub"`
+	TfClientConfig  *CreateClientProfileConfig  `json:"-" tf:"client_profile,sub"`
+	TfServerConfig  *CreateServerProfileConfig  `json:"-" tf:"server_profile,sub"`
+
+	ProfileType              string `json:"profileType" tf:"profile_type"`
+	FastTCPIdleTimeout       int    `json:"fastTcpIdleTimeout" tf:"fast_tcp_idle_timeout"`
+	FastUDPIdleTimeout       int    `json:"fastUdpIdleTimeout" tf:"fast_udp_idle_timeout"`
+	HTTPIdleTimeout          int    `json:"httpIdleTimeout" tf:"http_idle_timeout"`
```

Comment:
why these keys here ?

Reply:
For profiles also no need of json because these models CreateHttpProfileConfig only to achieve the profile type payloads from terraform later to send to cmp we are using same flat models so using these keys